Tests/DangerTests/DateFormatterHandlerTests.swift

```diff
@@ -0,0 +1,54 @@
+@testable import Danger
+import Foundation
+import XCTest
+
+class DateFormatterHandlerTests: XCTestCase {
+    func testRegularParseSuccess() throws {
+        let decoder = JSONDecoder()
+        decoder.dateDecodingStrategy = .custom(DateFormatter.dateFormatterHandler)
+
+        let json = """
+            { "date": "2019-04-10T21:57:45.000Z" }
+        """
+        let data = Data(json.utf8)
+
+        let dummy = try decoder.decode(DummyModel.self, from: data)
+        XCTAssertEqual(dummy.date, Date(timeIntervalSince1970: 1_554_933_465.000))
+    }
+
+    func testOnlyDateParseSuccess() throws {
+        let decoder = JSONDecoder()
+        decoder.dateDecodingStrategy = .custom(DateFormatter.dateFormatterHandler)
+
+        let json = """
+            { "date": "2019-04-10" }
+        """
+        let data = Data(json.utf8)
+
+        let dummy = try decoder.decode(DummyModel.self, from: data)
+        XCTAssertEqual(dummy.date, Date(timeIntervalSince1970: 1_554_854_400.000))
+    }
+
+    func testUnexpectedDateThrows() {
+        let decoder = JSONDecoder()
+        decoder.dateDecodingStrategy = .custom(DateFormatter.dateFormatterHandler)
+
+        let json = """
+            { "date": "2019*04*10" }
+        """
+        let data = Data(json.utf8)
+
+        do {
+            _ = try decoder.decode(DummyModel.self, from: data)
```

Comment:
You can use https://developer.apple.com/documentation/xctest/xctassertthrows here